lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -61,6 +61,11 @@ static llvm::cl::opt<bool> TFModuleLevelGraph(
         "verifying test outputs. Only set to false in unit tests,"
         "and when the unit tests do not involve function-typed attributes."));
 
+static llvm::cl::opt<bool> TFWarnSendRecv(
+    "tf-warn-send-recv", llvm::cl::init(false),
+    llvm::cl::desc(
+        "Emit warnings for sends/receives."));
+
 static llvm::cl::opt<bool> TFWarnScalarTransfer(
```

Comment:
do we want to remove `TFWarnScalarTransfer`, to reduce code complexity?

Reply:
It does not add too much code complexity as it is used in only two conditions. Also, it might be useful to keep this as a signal to think about warnings about scalar sends/receives when we revamp the code.   So, I will just leave it for now unless you feel strongly.

Reply:
It adds more test complexity (unit tests are being run with various flag combinations) that does not feel justified to me. I'd somewhat prefer to reduce such complexity, but am also fine with keeping it for now -- if the code/test complexity causes trouble, we can revisit?